File: 1556-make-two-arrays-equal-by-reversing-subarrays/make-two-arrays-equal-by-reversing-subarrays.py
class Solution:
    def canBeEqual(self, target: List[int], arr: List[int]) -> bool:
        mp = dict()
        for elem in target:
            mp[elem] = mp.get(elem, 0) + 1
        
        for elem in arr:
            if elem not in mp or mp[elem] == 0:
                return False
            mp[elem] -= 1
        
        return True

File: 1556-make-two-arrays-equal-by-reversing-subarrays/make-two-arrays-equal-by-reversing-subarrays.cpp
class Solution {
public:
    bool canBeEqual(vector<int>& target, vector<int>& arr) {
        unordered_map<int, int> dict;
        for (int elem : target){
            dict[elem]++;
        }
        for (int elem : arr){
            if (!dict.contains(elem) || dict[elem] == 0)
                return false;
            dict[elem]--;
        }
        return true;
    }
};